nomad/resource_namespace.go

```diff
@@ -0,0 +1,144 @@
+package nomad
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceNamespace() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNamespaceCreate,
+		Update: resourceNamespaceUpdate,
+		Delete: resourceNamespaceDelete,
+		Read:   resourceNamespaceRead,
+		Exists: resourceNamespaceExists,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Unique name for this namespace.",
+				Required:    true,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+			},
+
+			"description": {
+				Description: "Description for this namespace.",
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+
+			"quota": {
+				Description: "Quota to set for this namespace.",
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+		},
+	}
+}
+
+func resourceNamespaceCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	namespace := api.Namespace{
+		Name:        d.Get("name").(string),
+		Description: d.Get("description").(string),
+		Quota:       d.Get("quota").(string),
+	}
+
+	// upsert our namespace
+	log.Printf("[DEBUG] Creating namespace %q", namespace.Name)
+	_, err := client.Namespaces().Register(&namespace, nil)
+	if err != nil {
+		return fmt.Errorf("error inserting namespace %q: %s", namespace.Name, err.Error())
+	}
+	log.Printf("[DEBUG] Created namespace %q", namespace.Name)
+	d.SetId(namespace.Name)
+
+	return resourceNamespaceRead(d, meta)
+}
+
+func resourceNamespaceUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	namespace := api.Namespace{
+		Name:        d.Get("name").(string),
+		Description: d.Get("description").(string),
+		Quota:       d.Get("quota").(string),
+	}
+
+	// upsert our namespace
+	log.Printf("[DEBUG] Updating namespace %q", namespace.Name)
+	_, err := client.Namespaces().Register(&namespace, nil)
+	if err != nil {
+		return fmt.Errorf("error inserting namespace %q: %s", namespace.Name, err.Error())
+	}
+	log.Printf("[DEBUG] Updated namespace %q", namespace.Name)
+
+	return resourceNamespaceRead(d, meta)
+}
+
+func resourceNamespaceDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+	name := d.Id()
+
+	// delete the namespace
```

Comment:
_minor:_ I think we can remove this, since it's covered by the print statement below?